designer/server/src/main/scala/pl/touk/nussknacker/ui/process/newdeployment/DeploymentService.scala

```diff
@@ -157,7 +174,30 @@ class DeploymentService(
     ).leftMap(e => ConflictingDeploymentIdError(e.id))
   }
 
-  private def validateUsingDeploymentManager(
+  private def checkActiveScenariosLimits(
+      scenarioMetadata: ScenarioMetadata,
+      deployer: LoggedUser
+  ): EitherT[Future, RunDeploymentError, Unit] = {
+    implicit val loggedUser: LoggedUser = deployer
+    activeScenariosLimitProvider.forProcessingType(scenarioMetadata.processingType).flatten match {
+      case Some(ActiveScenariosLimit(activeScenariosLimit)) =>
+        EitherT {
+          scenarioStatusProvider
+            .getActiveScenariosCountFor(scenarioMetadata.processingType)
+            .map { activeScenariosCount =>
+              Either.cond(
+                test = activeScenariosCount < activeScenariosLimit,
```

Comment:
1. For me, a little bit more readable would be `currentlyActiveScenariosCount + 1 <= activeScenariosLimit` (thanks to that we emphasise that we check the simulated state after deployment
2.  What about `DeploymentUpdateStrategy`? Unfortunatelly we still don't have separated commands for deploy and redeploy, @gskrobisz is working on splitting them. For redeploy we should check `currentlyActiveScenariosCount <= activeScenariosLimit`